### ISSUE TITLE
Make gitpython to be extra dependency

### ISSUE THREAD

**Issue by radiophysicist** [TeoZosa/structlog-sentry-logger#812]
Hi,

While after merging of:
https://github.com/TeoZosa/structlog-sentry-logger/pull/775
the library code is working in case of missing `git` executable it is still impossible not to depend on `gitpython` library at all.

For me import of `structlog-sentry-logger` in my docker image (having `git` binary installed but not containing any git repos) causes the following error:
```
test_logger_1  | Traceback (most recent call last):
test_logger_1  |   File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
test_logger_1  |     return _run_code(code, main_globals, None,
test_logger_1  |   File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
test_logger_1  |     exec(code, run_globals)
test_logger_1  |   File "/src/app/__main__.py", line 5, in <module>
test_logger_1  |     from app.log import logger_factory
test_logger_1  |   File "/src/app/log.py", line 3, in <module>
test_logger_1  |     import structlog_sentry_logger
test_logger_1  |   File "/venv/lib/python3.10/site-packages/structlog_sentry_logger/__init__.py", line 4, in <module>
test_logger_1  |     from structlog_sentry_logger._config import (
test_logger_1  |   File "/venv/lib/python3.10/site-packages/structlog_sentry_logger/_config.py", line 61, in <module>
test_logger_1  |     ROOT_DIR = get_root_dir()
test_logger_1  |   File "/venv/lib/python3.10/site-packages/structlog_sentry_logger/_config.py", line 41, in get_root_dir
test_logger_1  |     return get_git_root()
test_logger_1  |   File "/venv/lib/python3.10/site-packages/structlog_sentry_logger/_config.py", line 56, in get_git_root
test_logger_1  |     git_root = git_repo.git.rev_parse("--show-toplevel")
test_logger_1  |   File "/venv/lib/python3.10/site-packages/git/cmd.py", line 741, in <lambda>
test_logger_1  |     return lambda *args, **kwargs: self._call_process(name, *args, **kwargs)
test_logger_1  |   File "/venv/lib/python3.10/site-packages/git/cmd.py", line 1315, in _call_process
test_logger_1  |     return self.execute(call, **exec_kwargs)
test_logger_1  |   File "/venv/lib/python3.10/site-packages/git/cmd.py", line 1109, in execute
test_logger_1  |     raise GitCommandError(redacted_command, status, stderr_value, stdout_value)
test_logger_1  | git.exc.GitCommandError: Cmd('git') failed due to: exit code(128)
test_logger_1  |   cmdline: git rev-parse --show-toplevel
test_logger_1  |   stderr: 'fatal: detected dubious ownership in repository at '/src'
test_logger_1  | To add an exception for this directory, call:
test_logger_1  | 
test_logger_1  | 	git config --global --add safe.directory /src'
```
making the library unusable even if I do not want to use any git repo inspection at all.

Seems to be logical to make  `gitpython` an extra dependency

**Comment by TeoZosa** [comment]
> For me import of `structlog-sentry-logger` in my docker image (having `git` binary installed but not containing any git repos) causes the following error:
> 
> ```
> ...
> ```
> making the library unusable even if I do not want to use any git repo inspection at all.
> 
> Seems to be logical to make `gitpython` an extra dependency

Hey @radiophysicist, thanks for raising the issue. I added a fix that should resolve this issue (#821) and cut a new release, could you try again with version [`0.20.1`](https://github.com/TeoZosa/structlog-sentry-logger/releases/tag/0.20.1)) and let me know if that resolves your issue?

Regarding your suggestion to make `gitpython` an extra dependency, that will definitely be something to keep in mind. Future work is planned around logger naming which will be the perfect time to revisit the use of `gitpython`.

**Comment by radiophysicist** [comment]
> > For me import of `structlog-sentry-logger` in my docker image (having `git` binary installed but not containing any git repos) causes the following error:
> > ```
> > ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > making the library unusable even if I do not want to use any git repo inspection at all.
> > Seems to be logical to make `gitpython` an extra dependency
> 
> Hey @radiophysicist, thanks for raising the issue. I added a fix that should resolve this issue (#821) and cut a new release, could you try again with version [`0.20.1`](https://github.com/TeoZosa/structlog-sentry-logger/releases/tag/0.20.1)) and let me know if that resolves your issue.
> 
> Regarding your suggestion to make `gitpython` an extra dependency, that will definitely be something to keep in mind. Future work is planned around logger naming which will be the perfect time to revisit the use of `gitpython`.

Thanks for a quick fix. 
I tried it and the crash due to a repository permissions seems to be resolved.

**Comment by TeoZosa** [comment]
> I tried it and the crash due to a repository permissions seems to be resolved.

Fantastic, glad to hear.
